File: frontend/src/types/category.ts
export interface Category {
  id: number
  name: string
  color: string
  userId: number
  createdAt: string
  updatedAt?: string
}

export interface CreateCategoryRequest {
  name: string
  color: string
}

export interface UpdateCategoryRequest {
  name?: string
  color?: string
}

export interface Tag {
  id: number
  name: string
  userId: number
  createdAt: string
}

export interface CreateTagRequest {
  name: string
}

File: backend/TodoApi/Models/Category.cs
namespace TodoApi.Models;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = "#000000"; // Hex color code
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Navigation property for many-to-many relationship
    public ICollection<TodoCategory> TodoCategories { get; set; } = new List<TodoCategory>();
}

